flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -123,9 +142,115 @@ public Builder serializer(BigQueryProtoSerializer serializer) {
+            FixedDelayRestartStrategyConfiguration strategy =
+                    (FixedDelayRestartStrategyConfiguration) restartStrategy;
+            if (strategy.getDelayBetweenAttemptsInterval().toMilliseconds()
+                            < MILLISECONDS_PER_SECOND
+                    || strategy.getRestartAttempts() > 10) {
+                throw new IllegalArgumentException(
+                        "Invalid restart strategy: FixedDelayRestartStrategyConfiguration should"
+                                + " be used with minimum restart delay 1 second,"
+                                + " and maximum 10 restart attempts.");
+            }
+        } else if (restartStrategy instanceof ExponentialDelayRestartStrategyConfiguration) {
+            ExponentialDelayRestartStrategyConfiguration strategy =
+                    (ExponentialDelayRestartStrategyConfiguration) restartStrategy;
+            if (strategy.getBackoffMultiplier() < 2.0
+                    || strategy.getInitialBackoff().toMilliseconds() < MILLISECONDS_PER_SECOND
+                    || strategy.getMaxBackoff().toMilliseconds() < (5L * MILLISECONDS_PER_MINUTE)
+                    || strategy.getResetBackoffThreshold().toMilliseconds()
+                            < MILLISECONDS_PER_HOUR) {
+                throw new IllegalArgumentException(
+                        "Invalid restart strategy: ExponentialDelayRestartStrategyConfiguration"
+                                + " should be used with minimum backoff multiplier 2,"
+                                + " minimum initial backoff 1 second, maximum backoff at-least"
+                                + " 5 minutes, and minimum reset threshold 1 hour");
+            }
+        } else if (restartStrategy instanceof FailureRateRestartStrategyConfiguration) {
+            FailureRateRestartStrategyConfiguration strategy =
+                    (FailureRateRestartStrategyConfiguration) restartStrategy;
+            double failureIntervalInMinutes =
+                    ((double) strategy.getFailureInterval().toMilliseconds())
+                            / MILLISECONDS_PER_MINUTE;
+            double allowedFailuresPerMinute =
+                    ((double) strategy.getMaxFailureRate()) / failureIntervalInMinutes;
+            if (strategy.getDelayBetweenAttemptsInterval().toMilliseconds()
+                            < MILLISECONDS_PER_SECOND
+                    || allowedFailuresPerMinute > 1.0) {
+                throw new IllegalArgumentException(
+                        "Invalid restart strategy: FailureRateRestartStrategyConfiguration should"
+                                + " be used with minimum restart delay 1 second, and maximum"
+                                + " failure rate of 1 per minute.");
+            }
+        } else if (restartStrategy instanceof NoRestartStrategyConfiguration) {
+            // This is fine! No validation needed.
```

Comment:
do we need to break here? i dont recall how java handles these cases

Reply:
Not needed. Added a debug log for legibility.